server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -350,11 +350,13 @@ void uploadMetadata(Collection<String> localSegmentsPostRefresh, SegmentInfos se
         userData.put(SequenceNumbers.MAX_SEQ_NO, Long.toString(maxSeqNo));
         segmentInfosSnapshot.setUserData(userData, false);
 
+        long translogGeneration = indexShard.getEngine().translogManager().getTranslogGeneration().translogFileGeneration;
         remoteDirectory.uploadMetadata(
             localSegmentsPostRefresh,
             segmentInfosSnapshot,
             storeDirectory,
-            indexShard.getOperationPrimaryTerm()
+            indexShard.getOperationPrimaryTerm(),
+            translogGeneration
```

Comment:
how are we guaranteeing that `translogGeneration` would always increase between two refreshes ? 

If we disable remote translog , it is not always going to increase . Even in case of async translog writes, refreshes can happen with same translog generation . Do we need a counter maintained in `RemoteRefreshListener` itself ?

Reply:
This is valid concern. Let me think more about it.

Reply:
thanks . should we use inverted timestamp ? this would enable us to search for `t - days` kind of use cases quicker .